# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmIonCompile.cpp
# Commit: 538920d2d777
# Full Hash: 538920d2d77751743a51f10cf730f1ce65b5d0ee
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2020-08-18 09:24:52
# Regressor Bug: 1561521
# File Overlap Count: 2
# Description:
#   Bug 1561521 - Rename TypeCode/ValType/RefType::Any to Extern. r=lth
#   
#   This commit renames the TypeCode/ValType/RefType occurrences of 'any' to 'extern'
#   to match what they actually represent. All public API facing occurrences have
#   already been switched to 'externref' so this should only be aligning the
# ==============================================================================

diff -r 3d3e3d124767 -r 538920d2d777 js/src/wasm/WasmIonCompile.cpp
--- a/js/src/wasm/WasmIonCompile.cpp	Tue Aug 18 00:03:29 2020 +0000
+++ b/js/src/wasm/WasmIonCompile.cpp	Tue Aug 18 00:04:37 2020 +0000
@@ -2598,7 +2598,7 @@
     case ValType::Ref:
       switch (value.type().refTypeKind()) {
         case RefType::Func:
-        case RefType::Any:
+        case RefType::Extern:
           MOZ_ASSERT(value.ref().isNull());
           result = f.nullRefConstant();
           break;
@@ -4674,7 +4674,7 @@
         if (!f.env().gcTypesEnabled()) {
           return f.iter().unrecognizedOpcode(&op);
         }
-        CHECK(EmitComparison(f, RefType::any(), JSOp::Eq,
+        CHECK(EmitComparison(f, RefType::extern_(), JSOp::Eq,
                              MCompare::Compare_RefOrNull));
 #endif
 #ifdef ENABLE_WASM_REFTYPES